integrationTests/mock/contracts/elrond.go

```diff
@@ -44,12 +52,71 @@ func (ec *ElrondContract) createContractFunctions() {
 	ec.AddHandler("getCurrentTxBatch", ec.getCurrentTxBatch)
 	ec.AddHandler("userRole", ec.userRole)
 	ec.AddHandler("getTokenIdForErc20Address", ec.getTokenIdForErc20Address)
+	ec.AddHandler("wasTransferActionProposed", ec.wasTransferActionProposed)
+	ec.AddHandler("proposeMultiTransferEsdtBatch", ec.proposeMultiTransferEsdtBatch)
+	ec.AddHandler("getActionSignerCount", ec.getActionSignerCount)
+	ec.AddHandler("proposeEsdtSafeSetCurrentTransactionBatchStatus", ec.proposeEsdtSafeSetCurrentTransactionBatchStatus)
 }
 
 func (ec *ElrondContract) getCurrentTxBatch(caller string, value string, arguments ...string) ([][]byte, error) {
 	log.Debug("getCurrentTxBatch", "caller", caller, "value", value, "arguments", arguments)
 
-	return make([][]byte, 0), nil
+	return make([][]byte, 0), nil //ec.currentTxBatch
+}
+
+func (ec *ElrondContract) wasTransferActionProposed(caller string, value string, arguments ...string) ([][]byte, error) {
+	log.Warn("wasTransferActionProposed", "caller", caller, "value", value, "arguments", fmt.Sprintf("%v", arguments))
+	batchId, _ := strconv.ParseInt(arguments[0], 10, 64)
+	ret := byte(1) //we assume we have the action in map
+	if _, ok := ec.action_mapper[big.NewInt(batchId)]; !ok {
```

Comment:
this map search doesn't work as expected. That is because the big int pointer is recreated each time. Why not make the map of type `map[int64]*big.Int`